luna.c

```diff
@@ -352,12 +351,9 @@ int doccrypt(void *inout, long in_size) {
 	/* As stored in tien_crypted_header below */
 	#define IVEC_BASE 0x6fe21307
 
-	if (   (r = DES_set_key_checked(&cbc1_key, &ks1))
-			|| (r = DES_set_key_checked(&cbc2_key, &ks2)) != 0
-			|| (r = DES_set_key_checked(&cbc3_key, &ks3)) != 0) {
-		printf("doccrypt - key error: %d\n", r);
-		return 1;
-	}
+	DES_set_key((DES_cblock*)&cbc1_key, &ks1);
```

Comment:
Yep, it was a basically a weak key check, in case the key was dynamically generated, which is not the case here.